clients/python/agentic-sandbox-client/agentic_sandbox/trace_manager.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2025 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import functools
+import logging
+import atexit
+import threading
+from contextlib import nullcontext
+
+# If optional dependency OpenTelemetry is not installed, define a complete set of mock objects
+# to prevent runtime errors.
+try:
```

Comment:
The optional import block with mock fallbacks is well-structured. It ensures the client remains lightweight for users who don't need tracing.